cloud/common.go

```diff
@@ -171,6 +171,32 @@ func (c Client) PostJSON(path, body string) ([]byte, error) {
 	return c.PostRaw(path, body, headers)
 }
 
+// DeleteRaw sends a DELETE request
+// Errors from the API are returned as an error
+func (c Client) DeleteRaw(path string) ([]byte, error) {
+	resp, err := c.Request("DELETE", path, "", nil)
+	if err != nil {
+		return nil, err
+	}
+	defer resp.Body.Close()
+
+	responseBody, err := io.ReadAll(resp.Body)
+	if err != nil {
+		return nil, err
+	}
+	// Pass the body to the callback function
+	c.rawFunc("DELETE", path, resp.StatusCode, responseBody)
+
+	// Convert the API's JSON error response to an error
+	if slices.Contains([]int{400, 401, 403, 404, 500}, resp.StatusCode) {
```

Comment:
Could this be done by try-convert to JSON, not to rely on some explicit return codes that might change?

Reply:
It could, but our openapi spec in osbuild-composer is supposed to be returning json for these.